src/reducers/historyReducer.js

```diff
@@ -102,9 +104,14 @@ export default function historyReducer(
         updatingTransaction: null,
       };
     case SET_GAS_INFO:
-      const gasPriceInfo = action.payload;
-      const isGasFetched = !!Object.keys(gasPriceInfo).length;
-      return { ...state, gasInfo: { gasPrice: gasPriceInfo, isFetched: isGasFetched } };
+      const { gasPrice, chain } = action.payload;
+      return {
+        ...state,
+        gasInfo: {
+          ...(state.gasInfo ?? {}),
```

Comment:
No need to do nil check for object spreading ;-) - only for arrays

```suggestion
          ...state.gasInfo,
```